woocommerce/woocommerce
# [Accessibility]Make label visible on the Filter by Attribute dropdown

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52084 .

The issue description says:

> If you want the ""slimmer"" design appearance, then you can style labels following the Material Design system (as the checkout page does) with a label that is positioned over the input and shifts up when content is entered in the field.

However, the dropdown input has a different behavior because the options fit in the input height:

![image](https://github.com/user-attachments/assets/8f5a5ac3-b2d3-453f-b457-1a1e09e42628)

Adding the label above and outside the input is not a good approach because it changes the UI and is a potential cause of regressions.

![image](https://github.com/user-attachments/assets/12a081e0-b99f-423b-93af-2a620b8c4805)

## The solution
The label was added to the dropdown, but it's only visible for screen readers, so it doesn't change the UI.
The visual indication of the input purpose can be achieved by leveraging the filter attribute title and updating it accordingly.

e.g
![image](https://github.com/user-attachments/assets/707c7f54-2a49-4a54-bb1c-36d1dc17aca2)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install a block-based theme.
2. On the Admin, go to Appearance > Editor > Templates > WooCommerce > Product Catalog.
3. Add the Filter by attribute block Inside the Product Collection block.
4. Select the "Size" option and click on "Done":
  ![image](https://github.com/user-attachments/assets/6789f4b8-1f48-4af4-86a8-eceba72af02e)
5. On the Settings panel, go to "Display style" and select "Dropdown".
  ![Screenshot 2024-12-10 at 15 30 14](https://github.com/user-attachments/assets/05221084-eb58-402e-935e-2e71afda38a8)
6. Save the template.
7. Go to the Shop page on the frontend.
8. Turn on screen reader.
9. Interact with the input and verify screen reader announces "Size" when the input is focused.
  ![image](https://github.com/user-attachments/assets/cd710238-42ef-46af-96c8-1aef0d52a8d5)




### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
